scripts/generate-artifacts-manifest/check_chart_images_support_multiple_architectures.py

```diff
@@ -0,0 +1,96 @@
+import json
+import sys
+import logging
+from pathlib import Path
+
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def load_json_file(file_path: str):
+    """Load and return the contents of a JSON file."""
+    try:
+        with open(file_path, "r", encoding="utf-8") as f:
+            return json.load(f)
+    except FileNotFoundError:
+        logging.error(f"File not found at {file_path}")
+        sys.exit(1)
+    except json.JSONDecodeError:
+        logging.error(f"Failed to decode JSON from {file_path}")
+        sys.exit(1)
+
+
+def parse_args(args):
+    """Parse and validate command-line arguments."""
+    if len(args) < 3:
+        logging.error("Usage: python check_images_support_multi_architectures.py <artifacts_manifest_path> <whitelisted_images_path>")
+        sys.exit(1)
+
+    artifacts_manifest_path = args[1]
+    whitelisted_images_path = args[2]
+
+    # Validate that files exist
+    if not Path(artifacts_manifest_path).is_file():
+        logging.error(f"Artifacts manifest file not found at {artifacts_manifest_path}")
+        sys.exit(1)
+    if not Path(whitelisted_images_path).is_file():
+        logging.error(f"Whitelisted images file not found at {whitelisted_images_path}")
+        sys.exit(1)
+
+    return artifacts_manifest_path, whitelisted_images_path
+
+
+def get_truefoundry_chart_images(artifacts_manifest):
+    """Extract the list of images associated with the 'truefoundry' chart."""
+    for artifact in artifacts_manifest:
+        if artifact.get("details", {}).get("chart") == "truefoundry":
```

Comment:
We need to include all truefoundry built charts. So this would include `truefoundry`, tfy-agent, tfy-notebook-controller, elasti etc